### DISCUSSION TITLE
Not showing ProductListPart and ProductImage field

### DISCUSSION THREAD

**Discussion by DotCat1985** [OrchardCMS/OrchardCore.Commerce#392] [Q&A] [answered]
Hi,
I added **OrchardCore.Commerce** module to my **OrchardCore.Cms.Web** project to add the e-commerce features.
Then, I installed them with **Orchard Core Commerce - Development** recipe, but:

- **ProductListPart** is not showing in create content type (OrchardCore.Lists has already been enabled)
- **ProductImage** field is not showing in create/edit product

How can we solve this issue?

**Comment by sarahelsaig** [comment]
Hi DotCat,

> ProductListPart is not showing in create content type

the `ProductListPart` is a content part, so it shouldn't show up in the list of content types. Check out [this tutorial](https://www.youtube.com/watch?v=NDUjn5_KdEM&list=PLuskKJW0FhJcSX7j0Bd-1X5hq3dgCtYwO&index=4) to learn more about Orchard Core's content model.
If you go to _Admin > Content > Content Definition > Content Types_ you can find the Product List content item (`~/Admin/ContentTypes/Edit/ProductList`). Here you can see that it's not "creatable" by default:
![image](https://github.com/OrchardCMS/OrchardCore.Commerce/assets/4007293/cd993da2-1b70-44c0-a264-93bc6f8c55f6)
It's because there is no point having more than one Product List content item, and one is already created by the _Orchard Core Commerce - Development_ recipe ([see here](https://github.com/OrchardCMS/OrchardCore.Commerce/blob/5a3043471307b91541ef171835722514512b61a8/src/Modules/OrchardCore.Commerce/Recipes/OrchardCore.Commerce.Samples.Product.recipe.json#L16), also in your site: `~/Admin/Contents/ContentItems/mainproductlist00000000000/Edit`). If you have to create a new one, you can edit the content type by ticking the Creatable checkbox mentioned above.

> ProductImage field is not showing in create/edit product

Could you give more details? I created a new site with the same recipe and _Product Image_ shows up for me in both existing and new content items:
![image](https://github.com/OrchardCMS/OrchardCore.Commerce/assets/4007293/4af0c762-3ee9-4bb4-9327-875c23c92e7a)

If you can't find it, please check if you see any errors in the log files in OrchardCore.Cms.Web/App_Data/logs.

**Reply by DotCat1985** [reply]
In my **OrchardCore.Cms.Web** project I created two tenants installed by using autosetup: "Default" and "AutoSetupTenant".
When I launch "AutoSetupTenant" first time, OrchardCore writes this log:

2023-12-26 07:51:07||0HN05RROOFFJU:00000005|Microsoft.AspNetCore.Antiforgery.DefaultAntiforgery|ERR|An exception was thrown while deserializing the token.
Microsoft.AspNetCore.Antiforgery.AntiforgeryValidationException: The antiforgery token could not be decrypted.
 ---> System.Security.Cryptography.CryptographicException: The key {05173e30-a72c-4aef-9dc1-032064c71c80} was not found in the key ring. For more information go to http://aka.ms/dataprotectionwarning
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.UnprotectCore(Byte[] protectedData, Boolean allowOperationsOnRevokedKeys, UnprotectStatus& status)
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.DangerousUnprotect(Byte[] protectedData, Boolean ignoreRevocationErrors, Boolean& requiresMigration, Boolean& wasRevoked)
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.Unprotect(Byte[] protectedData)
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenSerializer.Deserialize(String serializedToken)
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgery.GetCookieTokenDoesNotThrow(HttpContext httpContext)

**Reply by sarahelsaig** [reply]
That's normal due to autosetup. It's not related to OCC. I tried to create the setup you described, but I still can't reproduce your issue.
Just to be sure, are you using the latest package from [CloudSmith](https://cloudsmith.io/~orchardcore/repos/commerce/packages/detail/nuget/OrchardCore.Commerce/latest/)?

**Reply by DotCat1985** [reply]
I'm using 1.3.0 package from nuget.

https://www.nuget.org/packages/OrchardCore.Commerce/

**Reply by sarahelsaig** [reply]
I see, sorry about that. That's 4 months old (we only push to NuGet occasionally), and looks like those problems have been fixed since. Can you try [adding the CloudSmith feed](https://cloudsmith.io/~orchardcore/repos/commerce/setup/#formats-nuget) and then referencing the latest preview package (`1.3.1-preview-184`) in your project?

```pwsh
dotnet add package OrchardCore.Commerce -v 1.3.1-preview-184 -s https://nuget.cloudsmith.io/orchardcore/commerce/v3/index.json
```

**Comment by DotCat1985** [comment]
> I see, sorry about that. That's 4 months old (we only push to NuGet occasionally), and looks like those problems have been fixed since. Can you try [adding the CloudSmith feed](https://cloudsmith.io/~orchardcore/repos/commerce/setup/#formats-nuget) and then referencing the latest preview package (`1.3.1-preview-184`) in your project?
> 
> ```powershell
> dotnet add package OrchardCore.Commerce -v 1.3.1-preview-184 -s https://nuget.cloudsmith.io/orchardcore/commerce/v3/index.json
> ```

Sure. The command you suggest me has given the following error:

```
DotCat1985s-MacBook-Pro:OrchardCore.Cms.Web DotCat1985$ dotnet add package OrchardCore.Commerce -v 1.3.1-preview-184 -s https://nuget.cloudsmith.io/orchardcore/commerce/v3/index.json
  Determining projects to restore...
  Writing /var/folders/zm/5dfpsgls7vs1y7wrbnsmzn1h0000gn/T/tmpzCyh78.tmp
info : X.509 certificate chain validation will use the fallback certificate bundle at '/usr/local/share/dotnet/sdk/7.0.302/trustedroots/codesignctl.pem'.
info : X.509 certificate chain validation will use the fallback certificate bundle at '/usr/local/share/dotnet/sdk/7.0.302/trustedroots/timestampctl.pem'.
info : Adding PackageReference for package 'OrchardCore.Commerce' into project '/Users/DotCat1985/Documents/Visual Studio Code projects/orchardcore-webapp/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj'.
info : Restoring packages for /Users/DotCat1985/Documents/Visual Studio Code projects/orchardcore-webapp/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj...
info :   GET https://dl.cloudsmith.io/public/orchardcore/commerce/nuget/microsoft.visualstudio.setup.configuration.interop/index.json
info :   GET https://dl.cloudsmith.io/public/orchardcore/commerce/nuget/microsoft.sourcelink.github/index.json
info :   NotFound https://dl.cloudsmith.io/public/orchardcore/commerce/nuget/microsoft.sourcelink.github/index.json 265ms
info :   NotFound https://dl.cloudsmith.io/public/orchardcore/commerce/nuget/microsoft.visualstudio.setup.configuration.interop/index.json 309ms
error: NU1101: Unable to find package Microsoft.SourceLink.GitHub. No packages exist with this id in source(s): https://nuget.cloudsmith.io/orchardcore/commerce/v3/index.json
error: NU1101: Unable to find package Microsoft.VisualStudio.Setup.Configuration.Interop. No packages exist with this id in source(s): https://nuget.cloudsmith.io/orchardcore/commerce/v3/index.json
error: Package 'OrchardCore.Commerce' is incompatible with 'all' frameworks in project '/Users/DotCat1985/Documents/Visual Studio Code projects/orchardcore-webapp/src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj'.
```

**Reply by sarahelsaig** [reply]
The log says that it has only tried to find the package in the Microsoft.SourceLink.GitHub and Microsoft.VisualStudio.Setup.Configuration.Interop repositories. So I guess the Cloudsmith repo wasn't added. Did you follow the instructions on the page I linked?

**Comment by DotCat1985** [comment]
> I see, sorry about that. That's 4 months old (we only push to NuGet occasionally), and looks like those problems have been fixed since. Can you try [adding the CloudSmith feed](https://cloudsmith.io/~orchardcore/repos/commerce/setup/#formats-nuget) and then referencing the latest preview package (`1.3.1-preview-184`) in your project?
> 
> ```powershell
> dotnet add package OrchardCore.Commerce -v 1.3.1-preview-184 -s https://nuget.cloudsmith.io/orchardcore/commerce/v3/index.json
> ```



> The log says that it has only tried to find the package in the Microsoft.SourceLink.GitHub and Microsoft.VisualStudio.Setup.Configuration.Interop repositories. So I guess the Cloudsmith repo wasn't added. Did you follow the instructions on the page I linked?

Sorry, I refollowed those instructions and so I see both ProductList and ProductImage. Thank you!
When will the package **OrchardCore.Commerce 1.3.1** be released as stable release?

**Reply by sarahelsaig** [reply, marked as answer]
We made some major code changes related to payment processing since 1.3.0, so the next NuGet package (2.0.0) will contain breaking changes. We also have two issues on the horizon (#175 and #383) that may be breaking. So to minimize the pain, 2.0.0 will only come out when those two are resolved. 

There is no fixed release date. You can watch those issues in GH or follow the NuGet package's RSS feed (https://www.nuget.org/packages/OrchardCore.Commerce/atom.xml) to keep up to date on the next release.
